torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -201,9 +206,23 @@ std::shared_ptr<rpc::FutureMessage> DistEngine::runEngineAndAccumulateGradients(
           return;
         }
 
-        // Accumulate all the gradients in the context.
         TORCH_INTERNAL_ASSERT(grads.size() == outputEdges.size());
+
+        // Accumulate all the gradients in the context.
         for (size_t i = 0; i < grads.size(); i++) {
+          // Compute the number of references held to the same grad tensor that
```

Comment:
This wouldn't work as expected since grads[i] is const (since `const variable_list& grads`) and would anyways incur a copy as a result. The reason grads is const is because the Future holds a reference to the value and only allows callbacks to operate on a const ref of the value.